### ISSUE TITLE
Generating Charts

### ISSUE THREAD

**Issue by mirojs** [Azure-Samples/azure-search-openai-demo#1674]
> Please provide us with the following information:
> ---------------------------------------------------------------
The new model is good on data analysis. And suggestion on visualizing the results into typical charts?

### This issue is for a: (mark with an `x`)
```
- [ ] bug report -> please search issues before submitting
- [x] feature request
- [ ] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)
```

### Minimal steps to reproduce
>

### Any log messages given by the failure
>

### Expected/desired behavior
>

### OS and Version?
> Windows 7, 8 or 10. Linux (which distribution). macOS (Yosemite? El Capitan? Sierra?)

### azd version?
> run `azd version` and copy paste here.

### Versions
>

### Mention any other details that might be useful

> ---------------------------------------------------------------
> Thanks! We'll be in touch soon.

**Comment by Lawndemon** [comment]
I think you would need to include Dall-e if you want to have the solution return images. I'm planning to experiment with this once I get reranking working better but at this point I don't think it's a quick addition to put it mildly. Maybe it's easier than I'm thinking?

**Comment by pamelafox** [comment]
Another approach would be to use something like matplotlib to generate the charts. The models are pretty good at writing matplotlib code. You'd want to run the matplotlib code in a sandboxed environment through, like the new dynamic sessions on Azure Container Apps. See this example from @anthonychu's Pycon talk:
https://github.com/anthonychu/20240528-sessions-csv/blob/main/query-csv.ipynb

**Comment by anthonychu** [comment]
Yep that would work great here if the goal is to upload a CSV and ask it questions. My example doesn’t include a graph example but can add one.

**Comment by mirojs** [comment]
@anthonychu The OpenAI Spring video is quite impressive in terms of CSV visualization.
@pamelafox Matplotlib is good. Besides, it there any consideration of including semantic kernel etc. into the repo so that plugins could play a role?

**Comment by mirojs** [comment]
@anthonychu Could the graph example be added in the near future? Would Pandas be more convenient over Metaplotlib? Thanks
